lib/web_ui/lib/src/engine/bitmap_canvas.dart

```diff
@@ -108,15 +108,19 @@ class BitmapCanvas extends EngineCanvas {
   /// uses global transform of canvas to compute ratio.
   final double _density;
 
+  final RenderStrategy _renderStrategy;
+
   /// Allocates a canvas with enough memory to paint a picture within the given
   /// [bounds].
   ///
   /// This canvas can be reused by pictures with different paint bounds as long
   /// as the [Rect.size] of the bounds fully fit within the size used to
   /// initialize this canvas.
-  BitmapCanvas(this._bounds, {double density = 1.0})
+  BitmapCanvas(this._bounds, RenderStrategy renderStrategy,
```

Comment:
There's one place that passes a pre-existing render strategy, and 70 places that create a new instance. Maybe make it optional and default to a new instance in the initializer list?

Reply:
It is mostly tests. There are only 1 or 2 places in engine that create instance. Will likely make density required as well.